### PR TITLE
Fix OMSI Build for macOS

### PR DESCRIPTION
**Pull request by AnHeuermann**
Fix broken builds for macOS for compilation of OMSIBase and OMSISolver libs.

### CODE REVIEW

**Comment by AnHeuermann** [comment]
I fixed some first erros for OSX builds. For newer versions the version needs to get read in a different way from `sundials/sundials_config.h`. That should work now.

But for some reason my CMakeFile won't find the sundials version from OMCompiler/3rdParty
See
`FIND_PATH(SUNDIALS_INCLUDE_DIR NAMES "sundials/sundials_config.h" "kinsol/kinsol_dense.h" ${SUNDIALS_PATH}/include NO_DEFAULT_PATH)`
https://github.com/OpenModelica/OpenModelica/pull/248/files#diff-e587f515a8cf5da6783efb897945de80R55

Also RPATH is probably still wrong but the builds are crashing way earlier at the moment.

**Comment by sjoelund** [comment]
That's what my previous changes did for Sundials (look for my commits on CMakeLists.txt in old C++ runtime sources): FIND_PATH will search system libraries for some strange reason (even if NO_DEFAULT_PATH flag is given). The fix simply looked if a file exists, which works as expected on OSX.

**Comment by AnHeuermann** [comment]
I can't see whats the problem with this error message from the macOS test from Jenkins:
```
+ gmake -C OMCompiler/Compiler/Template/ -f Makefile.in OMC=/Users/hudson/jenkins-slave/ws/OpenModelica_PR-248/build/bin/omc

/Users/hudson/jenkins-slave/ws/OpenModelica_PR-248@tmp/durable-c9106ba9/script.sh: line 1: gmake: command not found

script returned exit code 127
```
@sjoelund Do you have an idea?

**Comment by sjoelund** [comment]
That seems odd indeed. The PATH variable was prepended. Maybe an `sh 'env | grep PATH'` in-between the commands is useful to see if that is somehow lost. Otherwise env.GMAKE could perhaps be set to "${env.MACPORTS}/bin/gmake"

File: .CI/common.groovy [hunk 1]
```diff
@@ -281,7 +282,7 @@ def tagName() {
 
 def makeCommand() {
   // OSX uses gmake as the GNU make program
-  return env.GMAKE ?: "make"
+  return env.GMAKE ?: "${env.MACPORTS}/bin/gmake"
```
Comment (sjoelund) [inline .CI/common.groovy]: This needs env.MACPORTS to be available rather than GMAKE

Comment (AnHeuermann) [reply .CI/common.groovy]: To be honest I'm a little lost with the Jenkins files. Could you try to fix the problem?
I will reset the latest commit.

**Comment by AnHeuermann** [comment]
There also seems to be a problem with the Win/MinGW pipeline:
https://test.openmodelica.org/jenkins/blue/organizations/jenkins/OpenModelica/detail/PR-248/31/pipeline#step-168-log-1

File: .CI/common.groovy (hunk 1 above)
Comment (sjoelund) [reply .CI/common.groovy]: I've been busy grading term papers and cleaning up before vacation. I'll change GMAKE on the slaves to the macports path instead.

Comment (AnHeuermann) [reply .CI/common.groovy]: Okay. Thanks a lot

**Comment by AnHeuermann** [comment]
@sjoelund or @adrpo Does one of you have time for a review?

**Comment by adrpo** [comment]
This is fine for me.

**Comment by AnHeuermann** [comment]
@adrpo Do you have an idea, why the msvc build for the C runtime doesn't work?
I added a check for the sundials libraries and added an error if they are not found.
See [OMCompiler/SimulationRuntime/c/CMakeLists.txt Line 55](https://github.com/OpenModelica/OpenModelica/pull/248/files?file-filters%5B%5D=.txt#diff-c092edbd51ac7cb8fdc96557f8611116R55)

I could remove this, but doesn't it mean that sundials is never found with MSVC?

**Comment by niklwors** [comment]
>

> 
> 
> @adrpo Do you have an idea, why the msvc build for the C runtime doesn't work?
> I added a check for the sundials libraries and added an error if they are not found.
> See [OMCompiler/SimulationRuntime/c/CMakeLists.txt Line 55](https://github.com/OpenModelica/OpenModelica/pull/248/files?file-filters%5B%5D=.txt#diff-c092edbd51ac7cb8fdc96557f8611116R55)
> 
> I could remove this, but doesn't it mean that sundials is never found with MSVC?
@AnHeuermann  The problem is the wrong path of OMCTRUNCHOME  in the CMakefile line 12. 
It should be:
SET(OMCTRUNCHOME ${CMAKE_CURRENT_SOURCE_DIR}/../../)

**Pull request merged**